packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreAliased.spec.ts

```diff
@@ -110,8 +105,17 @@ describeNoCompat("GC Data Store Aliased", (getTestObjectProvider) => {
         // Should be able to retrieve root datastore from remote
         const containerRuntime2 = mainDataStore2.containerRuntime;
         assert.doesNotThrow(async () => containerRuntime2.getRootDataStore(alias), "Aliased datastore should be root!");
+        const aliasableDataStore2 = await containerRuntime2.getRootDataStore(alias);
+        const aliasedDataStoreResponse2 = await aliasableDataStore2.request({url:"/"});
+        const aliasedDataStore2 = aliasedDataStoreResponse2.value as TestDataObject;
 
-        await summarizeOnContainer(container2);
-        // TODO: Check GC is notified
+        const gcReferences =
```

Comment:
We should not be working off of internal constructs of the garbage collector. This test will break as soon as the internals changes which should never happen. Also, that's not what end-to-end tests should do. They should test the end to end flow of a functionality.

What you can do is summarize before aliasing the data store, validate that the summary has this data store as unreferenced. Then summarize it again after aliasing and validate that the data stores is referenced.
The main validation here is that we don't get the unknown route error when summarizing before aliasing.

Reply:
We do cast to any to call bindToContext above but we don't really have a choice because we supported that pattern before and its still used.